src/pages/index.js

```diff
@@ -36,51 +39,123 @@ document.addEventListener("DOMContentLoaded", () => {
   const profileEditPopup = new PopupWithForm(
     "#profile-edit-modal",
     (formData) => {
-      userInfo.setUserInfo({
-        name: formData.heading,
-        job: formData.description,
-      });
+      profileEditPopup.renderLoading(true);
+      api
+        .setUserInfo({ name: formData.heading, about: formData.description })
+        .then(() => {
+          userInfo.setUserInfo({
+            name: formData.heading,
+            job: formData.description,
+          });
+          profileEditPopup.close();
+        })
+        .catch((err) => console.error(err))
+        .finally(() => profileEditPopup.renderLoading(false));
     }
   );
 
   const addCardPopup = new PopupWithForm("#add-card-modal", (formData) => {
+    addCardPopup.renderLoading(true);
     const cardData = { name: formData.heading, link: formData.description };
-    const cardElement = getCardElement(cardData);
-    section.addItem(cardElement);
+    api
+      .addCard(cardData)
+      .then((data) => {
+        const cardElement = getCardElement(data);
+        section.addItem(cardElement);
```

Comment:
Needs correction: There is an error in the console when I try to add a card `index.js:67 ReferenceError: section is not defined
    at index.js:64:9`

You should first define the section using `let` and then reassign it on line 143